File: SMSnmdscarm/SMSnmdscarm.R
# clear variables and close windows
rm(list=ls(all=TRUE))
graphics.off()

# install and load packages
libraries = c("MASS")
lapply(libraries, function(x) if (!(x %in% installed.packages())) {
    install.packages(x)})
lapply(libraries, library, quietly = TRUE, character.only = TRUE)

# variables
desc = c("Nissan","Kia","BMW","Audi")
#D = cbind(c(0,2,4,3),c(2,0,6,5),c(4,6,0,1),c(3,5,1,0))
D = cbind(c(0,2,5,3),c(2,0,6,4),c(5,6,0,1),c(3,4,1,0))
D

# main calculation
b = isoMDS(D, k = 2, maxit = 1000,tol=1e-10)

# 2D map
plot(b$points,xlab=expression(x[1]),ylab=expression(x[2]),xlim=c(-4,4),ylim=c(-4,4),main="nonmetric MDS",type="n")
text(b$points,desc,col="blue")

dist(b$points)

File: SMSnmdscarm/SMSnmdscarm.m
% clear variables and close windows
clear all
close all
clc

% variables
desc = cellstr(strvcat('Nissan','Kia','BMW','Audi'));
%D = cbind(c(0,2,4,3),c(2,0,6,5),c(4,6,0,1),c(3,5,1,0))
D = [[0,2,5,3]',[2,0,6,4]',[5,6,0,1]',[3,4,1,0]'];
D

% main calculation
[Y,stress,disparities] = mdscale(D,2,'criterion','stress')

% 2D map
plot(Y(:,1),Y(:,2),'wo')
xlabel('x_1','FontSize',16,'FontWeight','Bold')
ylabel('x_2','FontSize',16,'FontWeight','Bold')
xlim([-4,4])
ylim([-4,4])
title('nonmetric MDS','FontSize',16,'FontWeight','Bold')
text(Y(:,1),Y(:,2),desc,'color','b','FontSize',16,'FontWeight','Bold')
box on
set(gca,'LineWidth',1.6,'FontSize',16,'FontWeight','Bold')

dist(Y)